### PR TITLE
 Added Blip Types and Checkpoint Types,  Added Async C# API & Examples 

### PR DESCRIPTION
**Pull request by DasChaosAT**

### CODE REVIEW

**Comment by FabianTerhorst** [comment]
Maybe add async code like an Task.Delay in the async event handlers to indicate there usage.

**Comment by DasChaosAT** [comment]
Okay, will do that tomorrow

File: docs/Resources/Checkpoints.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Checkpoints
+
+## Checkpoint List
+
+| Preview                                 | Checkpoint                  | ID   |
+| ----------------------------------------| --------------------------- | ---- |
+| ![](checkpoint_img/01.jpg ':size=250')  | CylinderSingleArrow         | 1    |
```
Comment (FabianTerhorst) [inline docs/Resources/Checkpoints.md]: Checkpoint ids are starting with 1

File: docs/Resources/Blips/Types.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+## Blip Types
+
+| Type                        | ID   |
+| --------------------------- | ---- |
+| Vehicle                     | 1    |
+| Ped		                  | 2    |
+| Enemy                       | 2    |
```
Comment (FabianTerhorst) [inline docs/Resources/Blips/Types.md]: Stay with Ped and remove duplicate

**Comment by vadz1999** [comment]
Remove duplicated type 2 in blip types (keep only ped), checkpoint types should start with 0

**Comment by DasChaosAT** [comment]
Should be fixed

File: docs/ServerAPI/csharp/Event_Examples_Async.md [hunk 3]
```diff
@@ -0,0 +1,546 @@
...
+            AltAsync.OnVehicleChangeSeat += OnVehicleChangeSeat;
+            AltAsync.OnVehicleEnter += OnVehicleEnter;
+            AltAsync.OnVehicleLeave += OnVehicleLeave;
+            AltAsync.OnPlayerEvent += OnPlayerEvent;
+
+            // Custom Event Triggers can be found at the bottom
+            // Custom Event Type 01
+            AltAsync.On("customEvent01", args =>
+            {
+                AltAsync.Log("customEvent01 triggered with following arguments: " + args[0]);
```
Comment (FabianTerhorst) [inline docs/ServerAPI/csharp/Event_Examples_Async.md]:
```async args => 
{
await Task.Delay(1000)
AltAsync.Log("customEvent01 triggered with following arguments: " + args[0]);
}```

**Comment by DasChaosAT** [comment]
Added async to custom events

**Comment by FabianTerhorst** [comment]
@vadz1999 are you happy now with the enum values?

**Pull request merged**